pyathena/model.py

```diff
@@ -241,6 +241,158 @@ def result_reuse_minutes(self) -> Optional[int]:
         return self._result_reuse_minutes
 
 
+class AthenaCalculationExecutionStatus:
+    STATE_CREATING: str = "CREATING"
+    STATE_CREATED: str = "CREATED"
+    STATE_QUEUED: str = "QUEUED"
+    STATE_RUNNING: str = "RUNNING"
+    STATE_CANCELING: str = "CANCELING"
+    STATE_CANCELLED: str = "CANCELLED"
+    STATE_COMPLETED: str = "COMPLETED"
+    STATE_FAILED: str = "FAILED"
+
+    def __init__(self, response: Dict[str, Any]) -> None:
+        status = response.get("Status", None)
+        if not status:
+            raise DataError("KeyError `Status`")
+        self._state: Optional[str] = status.get("State", None)
+        self._state_change_reason: Optional[str] = status.get("StateChangeReason", None)
+        self._submission_date_time: Optional[datetime] = status.get("SubmissionDateTime", None)
+        self._completion_date_time: Optional[datetime] = status.get("CompletionDateTime", None)
+
+        statistics = response.get("Statistics", None)
+        if not statistics:
+            raise DataError("KeyError `Statistics`")
+        self._dpu_execution_in_millis: Optional[int] = statistics.get("DpuExecutionInMillis", None)
+        self._progress: Optional[str] = statistics.get("Progress", None)
+
+    @property
+    def state(self) -> Optional[str]:
+        return self._state
+
+    @property
+    def state_change_reason(self) -> Optional[str]:
+        return self._state_change_reason
+
+    @property
+    def submission_date_time(self) -> Optional[datetime]:
+        return self._submission_date_time
+
+    @property
+    def completion_date_time(self) -> Optional[datetime]:
+        return self._completion_date_time
+
+    @property
+    def dpu_execution_in_millis(self) -> Optional[int]:
+        return self._dpu_execution_in_millis
+
+    @property
+    def progress(self) -> Optional[str]:
+        return self._progress
+
+
+class AthenaCalculationExecution(AthenaCalculationExecutionStatus):
+    def __init__(self, response: Dict[str, Any]) -> None:
+        super(AthenaCalculationExecution, self).__init__(response)
+
+        self._calculation_id: Optional[str] = response.get("CalculationExecutionId", None)
```

Comment:
nitpick: You can simply leave this as `response.get("CalculationExecutionId")` since `.get` by default returns None if not found

Reply:
Fixed: https://github.com/laughingman7743/PyAthena/pull/497/commits/8a6202bd3a4a7f6c4fc1b4ba0ccee22eab756525

Reply:
Fixed: https://github.com/laughingman7743/PyAthena/pull/497/commits/6529f5dfb6dfee601321c7fa5c6b28d264ca3d14